src/Microsoft.AspNetCore.Rewrite/Internal/ConditionEvaluator.cs

```diff
@@ -24,18 +24,26 @@ public static MatchResults Evaluate(IEnumerable<Condition> conditions, RewriteCo
                     continue;
                 }
 
-                prevCond = condition.Evaluate(context, ruleMatch, prevCond);
+                var condResult = condition.Evaluate(context, ruleMatch, prevResult);
 
                 if (condition.OrNext)
                 {
-                    orSucceeded = prevCond.Success;
+                    orSucceeded = condResult.Success;
                 }
-                else if (!prevCond.Success)
+                else if (!condResult.Success)
                 {
-                    return prevCond;
+                    return condResult;
                 }
+
+                if (condResult.Success && trackAllCaptures && prevResult?.BackReferences != null)
+                {
+                    prevResult.BackReferences.Add(condResult.BackReferences);
+                    condResult.BackReferences = prevResult.BackReferences;
```

Comment:
Hmm, this may end up setting MatchResult.EmptySuccess. Should change this so the loop tracks backrefrences independent of MatchResult. In fact, there's no reason to hold on to the prior match result, only the backrefrences. Change Evaluate to take in references rather than results.